packages/swapper/src/swappers/osmosis/OsmosisSwapper.ts

```diff
@@ -0,0 +1,317 @@
+    const { sellAsset, buyAsset, sellAmount, sellAssetAccountNumber, receiveAddress } = trade
+
+    const pair = sellAsset.symbol + '_' + buyAsset.symbol
+    const isFromOsmo = pair === 'OSMO_ATOM'
+    const sellAssetDenom = symbolDenomMapping[sellAsset.symbol as keyof SymbolDenomMapping]
+    const buyAssetDenom = symbolDenomMapping[buyAsset.symbol as keyof SymbolDenomMapping]
+    let ibcSellAmount
+
+    const osmosisAdapter = this.deps.adapterManager.get(osmosisChainId) as
+      | osmosis.ChainAdapter
+      | undefined
+    const cosmosAdapter = this.deps.adapterManager.get(cosmosChainId) as
+      | cosmos.ChainAdapter
+      | undefined
+
+    if (cosmosAdapter && osmosisAdapter) {
+      let sellAddress
+
+      if (!isFromOsmo) {
+        const sellBip44Params = cosmosAdapter.buildBIP44Params({
+          accountNumber: Number(sellAssetAccountNumber)
+        })
+
+        sellAddress = await cosmosAdapter.getAddress({ wallet, bip44Params: sellBip44Params })
+
+        if (!sellAddress) throw Error('Failed to get atomAddress!')
+
+        const transfer = {
+          sender: sellAddress,
+          receiver: receiveAddress,
+          amount: String(sellAmount)
+        }
+
+        const { tradeId } = await performIbcTransfer(
+          transfer,
+          cosmosAdapter,
+          wallet,
+          this.deps.cosmosUrl,
+          this.deps.osmoUrl,
+          'uatom',
+          'channel-141',
+          '0'
+        )
+
+        // wait till confirmed
+        const pollResult = await pollForComplete(tradeId, this.deps.cosmosUrl)
+        if (pollResult !== 'success') throw new Error('ibc transfer failed')
+
+        ibcSellAmount = await pollForAtomChannelBalance(receiveAddress, this.deps.osmoUrl)
+      } else if (isFromOsmo) {
+        const sellBip44Params = osmosisAdapter.buildBIP44Params({
+          accountNumber: Number(sellAssetAccountNumber)
+        })
+        sellAddress = await osmosisAdapter.getAddress({ wallet, bip44Params: sellBip44Params })
+
+        if (!sellAddress) throw Error('Failed to get osmoAddress!')
+      } else {
+        throw Error('Pair not supported! ' + pair)
+      }
+      const osmoAddress = isFromOsmo ? sellAddress : receiveAddress
+      const accountUrl = `${this.deps.osmoUrl}/auth/accounts/${osmoAddress}`
+      const responseAccount = await (async () => {
+        try {
+          return axios.get(accountUrl)
+        } catch (e) {
+          throw new Error(`Failed to get account: ${e}`)
+        }
+      })()
+      const accountNumber = responseAccount.data.result.value.account_number || 0
+      const sequence = responseAccount.data.result.value.sequence || 0
+
+      const osmoAddressNList = bip32ToAddressNList("m/44'/118'/0'/0/0")
+
+      const tx1 = {
+        memo: '',
+        fee: {
+          amount: [
+            {
+              amount: FEE.toString(),
+              denom: 'uosmo'
+            }
+          ],
+          gas: GAS
```

Comment:
In relation to fees and gas, we should leverage `chainAdapter.getFeeData` instead of creating a separate set of constants in swapper